### PR TITLE
Refactor/traffic simulator/spawning

### PR DESCRIPTION
**Pull request by yamacir-kit**
## Types of PR

- [ ] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

### CODE REVIEW

File: openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp [hunk 1]
```diff
@@ -28,9 +28,15 @@ TeleportAction::TeleportAction(const pugi::xml_node & node, Scope & scope)
 {
 }
 
-auto TeleportAction::accomplished() noexcept -> bool { return true; }
+auto TeleportAction::accomplished() noexcept -> bool  //
```
Comment (hakuturu583) [inline openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp:31]: Why did you edited this line?

File: openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp [hunk 2]
```diff
@@ -28,9 +28,15 @@ TeleportAction::TeleportAction(const pugi::xml_node & node, Scope & scope)
...
+{
+  return true;
+}
 
-auto TeleportAction::endsImmediately() noexcept -> bool { return true; }
+auto TeleportAction::endsImmediately() noexcept -> bool  //
```
Comment (hakuturu583) [inline openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp:36]: Why did you edited this line?

Comment (yamacir-kit) [reply openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp]: 05188d8 fixes this.

File: openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp (hunk 1 above)
Comment (yamacir-kit) [reply openscenario/openscenario_interpreter/src/syntax/teleport_action.cpp]: 05188d8 fixes this.

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp [hunk 3]
```diff
@@ -34,9 +34,27 @@ namespace entity
...
+      return name;
+    }
+
+    static auto context_gamma() noexcept -> const std::string &
+    {
+      static const std::string name = "TODO";
+      return name;
+    }
+
+    static auto default_behavior() noexcept -> const std::string & { return behavior_tree(); }
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:51]: Can you follow section 4.4 and 4.5 in this coding guidelines?

http://wiki.ros.org/CppStyleGuide

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp [hunk 4]
```diff
@@ -34,9 +34,27 @@ namespace entity
...
+  };
+
+  explicit PedestrianEntity(
+    const std::string & name,                                   //
+    const traffic_simulator_msgs::msg::PedestrianParameters &,  //
+    const std::string & = BuiltinBehavior::default_behavior());
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:57]: Can you follow section 4.4 and 4.5 in this coding guidelines?

http://wiki.ros.org/CppStyleGuide

File: simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp [hunk 5]
```diff
@@ -37,9 +37,27 @@ namespace entity
...
+    }
+
+    static auto context_gamma() noexcept -> const std::string &
+    {
+      static const std::string name = "TODO";
+      return name;
+    }
+
+    static auto default_behavior() noexcept -> const std::string & { return behavior_tree(); }
+  };
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp:55]: Can you follow section 4.4 and 4.5 in this coding guidelines?

http://wiki.ros.org/CppStyleGuide

File: simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp [hunk 6]
```diff
@@ -37,9 +37,27 @@ namespace entity
...
+
+  explicit VehicleEntity(
+    const std::string & name,                                //
+    const traffic_simulator_msgs::msg::VehicleParameters &,  //
+    const std::string & = BuiltinBehavior::default_behavior());
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp:60]: Can you follow section 4.4 and 4.5 in this coding guidelines?

http://wiki.ros.org/CppStyleGuide

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp (hunk 4 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:57]: I think 
```
BuiltinBehavior::defaultBehavior()
```

is right.

File: simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp (hunk 6 above)
Comment (yamacir-kit) [reply simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp]: 1bd92c2 fixed this.

File: simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp (hunk 5 above)
Comment (yamacir-kit) [reply simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp]: 1bd92c2 fixed this.

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp (hunk 4 above)
Comment (yamacir-kit) [reply simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp]: 1bd92c2 fixed this.

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp (hunk 3 above)
Comment (yamacir-kit) [reply simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp]: 1bd92c2 fixed this.

**Review by hakuturu583 — APPROVED** [review]

**Pull request merged**